DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorTrackSoA.cc

```diff
@@ -0,0 +1,117 @@
+// -*- C++ -*-
+///bookLayer
+// Package:    SiPixelPhase1MonitorTrackSoA
+// Class:      SiPixelPhase1MonitorTrackSoA
+//
+/**\class SiPixelPhase1MonitorTrackSoA SiPixelPhase1MonitorTrackSoA.cc 
+*/
+//
+// Author: Suvankar Roy Chowdhury
+//
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+// DQM Histograming
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "CUDADataFormats/Track/interface/PixelTrackHeterogeneous.h"
+
+class SiPixelPhase1MonitorTrackSoA : public DQMEDAnalyzer {
+public:
+  explicit SiPixelPhase1MonitorTrackSoA(const edm::ParameterSet&);
+  ~SiPixelPhase1MonitorTrackSoA() override;
+  void bookHistograms(DQMStore::IBooker& ibooker, edm::Run const& iRun, edm::EventSetup const& iSetup) override;
+  void analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  edm::EDGetTokenT<PixelTrackHeterogeneous> tokenSoATrack_;
+  std::string topFolderName_;
+  MonitorElement* hnTracks;
+  MonitorElement* hnHits;
+  MonitorElement* hchi2;
+  MonitorElement* hpt;
+  MonitorElement* heta;
+  MonitorElement* hphi;
+  MonitorElement* hz;
+};
+
+//
+// constructors
+//
+
+SiPixelPhase1MonitorTrackSoA::SiPixelPhase1MonitorTrackSoA(const edm::ParameterSet& iConfig) {
+  tokenSoATrack_ = consumes<PixelTrackHeterogeneous>(iConfig.getParameter<edm::InputTag>("pixelTrackSrc"));
+  topFolderName_ = iConfig.getParameter<std::string>("TopFolderName");  //"SiPixelHeterogeneous/PixelTrackSoA";
+}
+
+SiPixelPhase1MonitorTrackSoA::~SiPixelPhase1MonitorTrackSoA() {
+  // do anything here that needs to be done at desctruction time
+  // (e.g. close files, deallocate resources etc.)
+  edm::LogInfo("SiPixelPhase1MonitorTrackSoA") << ">>> Destroy SiPixelPhase1MonitorTrackSoA ";
```

Comment:
is it really needed? 
This has a cost in production as the message is constructed anyway but dropped.